### PR TITLE
Add soft-commit function

### PR DESCRIPTION
**Pull request by giteshnandre**
* Added soft commit function. 
* Added test. 
* Added test for existing commit function too.

### CODE REVIEW

File: lib/rsolr/client.rb [hunk 1]
```diff
@@ -122,6 +122,14 @@ def commit opts = {}
     update opts.merge(:data => builder.commit( commit_attrs ))
   end
 
+  # soft commit
+  #
+  # https://lucene.apache.org/solr/guide/updatehandlers-in-solrconfig.html#commit-and-softcommit
+  #
+  def soft_commit opts = {}
+    commit(opts.merge params: { softCommit: true })
```
Comment (hackartisan) [inline lib/rsolr/client.rb:130]: `softCommit` doesn't appear to be an option any more for a commit update. I think the way to do a soft commit is via `waitSearcher: true`

The docs linked in the comment are for solr configuration, but the functionality in the code uses a POST update. See
: https://solr.apache.org/guide/uploading-data-with-index-handlers.html#commit-and-optimize-during-updates for options at update time.

Comment (tpendragon) [reply lib/rsolr/client.rb:130]: @hackartisan I think softCommit is still a thing - there's just also hard commits where you don't open a searcher.

**Review by tpendragon — APPROVED** [review]

File: lib/rsolr/client.rb (hunk 1 above)
Comment (hackartisan) [reply lib/rsolr/client.rb:130]: @tpendragon yes I see now. @giteshnandre sorry for my mis-read of the documentation.

Do either of you know how soft commits interact with searcher initialization? I'm curious if a soft commit update blocks until a new searcher is initialized. Neither of these 2 documentation pages mention both.

Comment (tpendragon) [reply lib/rsolr/client.rb:130]: Soft commit definitely blocks until a new searcher is initialized.

**Comment by jrochkind** [comment]
It's basically just giving you a `soft_commit` method which is just a convenience to add `softCommit=true` param on to the request?

**Review by bess — APPROVED** [review]

**Pull request merged**